system/lib/pthread/pthread_create.c

```diff
@@ -116,6 +116,12 @@ int __pthread_create(pthread_t* restrict res,
   new->tsd = malloc(PTHREAD_KEYS_MAX * sizeof(void*));
   memset(new->tsd, 0, PTHREAD_KEYS_MAX * sizeof(void*));
 
+  if (attrp && attrp != __ATTRP_C11_THREAD && attrp->_a_detach) {
```

Comment:
This logic looks different than before - it also checks `attrp->_a_detach`?

Reply:
The old code is doing the same thing is just a little harder to read:

```
   var initialState = {{{ cDefine('DT_JOINABLE') }}};
    // When musl creates C11 threads it passes __ATTRP_C11_THREAD (-1) which
    // treat as if it was NULL.
    if (attr && attr != {{{ cDefine('__ATTRP_C11_THREAD') }}}) {
      ...
      if ({{{ makeGetValue('attr', 12/*_a_detach*/, 'i32') }}}) {
        initialState = {{{ cDefine('DT_DETACHED') }}};
      }
```

Reply:
Ah, right, I see it now.